File: backend/KanbanApi/Models/KanbanColumn.cs
using System.ComponentModel.DataAnnotations;

namespace KanbanApi.Models;

public class KanbanColumn
{
    public int Id { get; set; }

    [Required]
    public string Name { get; set; } = string.Empty;

    public int Position { get; set; }

    public int UserId { get; set; }
    public User? User { get; set; }

    public List<Ticket> Tickets { get; set; } = new();
}

File: frontend/kanban-frontend/src/app/core/models/kanban-column.ts
export interface KanbanColumn {
  id: number;
  name: string;
  position: number;
  userId: number;
}

export interface CreateKanbanColumnDto {
  name: string;
  position: number;
  userId: number;
}

export interface UpdateKanbanColumnDto {
  name: string;
  position: number;
  userId: number;
}
